### PR TITLE
Backend applicant name

### PR DESCRIPTION
**Pull request by floresj**
**Notes**
- Updated the validate api design docs to render `json` appropriately
- Renamed `PersonField` to `NameField` to store a persons name information
- Added validation logic for firstname, middlename, lastname and suffix fields
- Added a generic name field that performs basic name validations for catching invalid characters
- Added `/validate/applicant/name` endpoint

### CODE REVIEW

File: api/model/form/name_field.go [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+	Middle MiddlenameField
+	Suffix SuffixField
+}
+
+// Valid ensures that a person name is valid
+func (f NameField) Valid() (bool, error) {
+	var stack ErrorStack
+
+	if ok, err := f.First.Valid(); !ok {
+		stack.Append("Firstname", err)
```
Comment (bmallred) [inline api/model/form/name_field.go]: Should this come back as `First` to match what is passed in?

File: api/design/validate.md [hunk 2]
```diff
@@ -206,5 +214,57 @@ FORMAT: 1A
...
+
+        X-Eqip-Media-Type: eqip.v1
+
+    + Body
+
+```json
+    {
+      "Errors": [
+        {
+          "Fieldname": "Firstname",
```
Comment (bmallred) [inline api/design/validate.md]: Depending on the conversation in `name_field.go` this may be changed to `First`

File: api/design/validate.md [hunk 3]
```diff
@@ -206,5 +214,57 @@ FORMAT: 1A
...
+
+# Validate Applicant Name [/validate/applicant/name]
+
+## Validates a persons name entry [POST]
+
++ Request
+    + Headers
+
+        Authorization: Bearer
+        Accept: application/json
```
Comment (bmallred) [inline api/design/validate.md]: We should probably add a sample of what the request body may appear as. Right now it is the following:

```json
{
  "Last": "Xavier",
  "First": "Charles",
  "Middle": "F",
  "Suffix": "Other",
  "SuffixOther": "XV"
}
```

File: api/design/validate.md [hunk 4]
```diff
@@ -206,5 +214,57 @@ FORMAT: 1A
...
+          "Fieldname": "Suffix",
+          "Errors": [
+            {
+              "Fieldname": "Suffix",
+              "Error": ""
+            },
+            {
+              "Fieldname": "SuffixOther",
+              "Error": ""
+            }
```
Comment (bmallred) [inline api/design/validate.md]: We may need to talk about whether this should be another `object`. Currently we are passing in `Suffix` and `SuffixOther` in a flat nature. Should we pass it back the same?

File: api/model/form/name_field.go (hunk 1 above)
Comment (floresj) [inline api/model/form/name_field.go]: Ahh yes, good catch. I will change that up.

File: api/design/validate.md (hunk 3 above)
Comment (floresj) [inline api/design/validate.md]: Ok, will add those in.

File: api/design/validate.md (hunk 4 above)
Comment (floresj) [inline api/design/validate.md]: I forgot that it was being sent separately. I will adjust that so they are separate and the logic occurs in the `NameField.go`

**Pull request merged**